examples/pytorch/deepwalk/model.py

```diff
@@ -0,0 +1,430 @@
+        self.u_embeddings.weight.share_memory_()
+        self.v_embeddings.weight.share_memory_()
+        if self.adam:
+            self.state_sum_u.share_memory_()
+            self.state_sum_v.share_memory_()
+
+    def set_device(self, gpu_id):
+        self.device = torch.device("cuda:%d" % gpu_id)
+        print("The device is", self.device)
+        self.lookup_table = self.lookup_table.to(self.device)
+        self.index_emb_posu = self.index_emb_posu.to(self.device)
+        self.index_emb_posv = self.index_emb_posv.to(self.device)
+        if self.fast_neg:
+            self.index_emb_negu = self.index_emb_negu.to(self.device)
+            self.index_emb_negv = self.index_emb_negv.to(self.device)
+        self.grad_u = self.grad_u.to(self.device)
+        self.grad_v = self.grad_v.to(self.device)
+        if self.avg_sgd:
+            self.grad_avg = self.grad_avg.to(self.device)
+
+    def all_to_device(self, gpu_id):
+        """ move all of the parameters to a single GPU """
+        self.device = torch.device("cuda:%d" % gpu_id)
+        self.set_device(gpu_id)
+        self.u_embeddings = self.u_embeddings.cuda(gpu_id)
+        self.v_embeddings = self.v_embeddings.cuda(gpu_id)
+        if self.adam:
+            self.state_sum_u = self.state_sum_u.to(self.device)
+            self.state_sum_v = self.state_sum_v.to(self.device)
+
+    def fast_sigmoid(self, score):
+        """ do fast sigmoid by looking up in a defined table """
+        idx = torch.floor((score + 6.01) / 0.01).long()
+        return self.lookup_table[idx]
+
+    def fast_learn_super(self, batch_walks, lr, neg_nodes=None):
+        """ Learn a batch of random walks in a fast way.
+        For each positive/negative node pair (i,j), the updating procedure is as following:
+        score = self.fast_sigmoid(u_embedding[i].dot(v_embedding[j]))
+        # label = 1 for positive samples; label = 0 for negative samples.
+        u_embedding[i] += (label - score) * v_embedding[j]
+        v_embedding[i] += (label - score) * u_embedding[j]
+
+        Parameters
+        ----------
+        batch_walks list : a list of node sequnces
+        lr float : current learning rate
+        neg_nodes torch.LongTensor : a long tensor of sampled true negative nodes. If neg_nodes is None,
+            then do negative sampling randomly from the nodes in batch_walks as an alternative.
+
+        Usage example
+        -------------
+        batch_walks = [torch.LongTensor([1,2,3,4]), 
+                       torch.LongTensor([2,3,4,2])])
+        lr = 0.01
+        neg_nodes = None
+        """
+        if self.adam:
+            lr = self.lr
+
+        # [batch_size, walk_length]
+        nodes = torch.stack(batch_walks)
+        if self.only_gpu:
+            nodes = nodes.to(self.device)
+            if neg_nodes is not None:
+                neg_nodes = neg_nodes.to(self.device)
+        emb_u = self.u_embeddings.weight[nodes].view(-1, self.emb_dimension).to(self.device)
```

Comment:
you can use

emb_u = self.u_embeddings(nodes).view(-1, self.emb_dimension).to(self.device)
emb_v = self.v_embeddings(nodes).view(-1, self.emb_dimension).to(self.device)